exchanges/graphcache/src/operations/query.ts

```diff
@@ -90,20 +90,23 @@ export const read = (
     pushDebugNode(rootKey, operation);
   }
 
-  let data: Data | undefined = input || ({} as Data);
-  data =
+  // NOTE: This may reuse "previous result data" as indicated by the
+  // `originalData` argument in readRoot(). This behaviour isn't used
+  // for readSelection() however, which always produces results from
+  // scratch
+  const data =
     rootKey !== ctx.store.rootFields['query']
-      ? readRoot(ctx, rootKey, rootSelect, data)
-      : readSelection(ctx, rootKey, rootSelect, data);
+      ? readRoot(ctx, rootKey, rootSelect, input || ({} as Data))
+      : readSelection(ctx, rootKey, rootSelect, {} as Data);
```

Comment:
This is the relevant line for review :)

Reply:
@JoviDeCroock Merging as a hotfix; the review should be simple, but @sarmeyer has tested this and it works just fine 🙌